### PR TITLE
Chapter 18: Working with static table views, camera and NSLayout Constraint

### PR DESCRIPTION
**Pull request by AndreeaSburlea**
## New features
- static table views
- text field and text view
- barbuttons

## Changes in code
- displaying the photo library
- difining auto layout constrains

## Screenshot
Screenshot 2022-05-06 at 14 49 22
Screenshot 2022-05-06 at 14 49 04
Screenshot 2022-05-06 at 14 48 47

### CODE REVIEW

File: FoodPin/FoodPin/Controller/NewRestaurantController.swift [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+            alert.addAction(UIAlertAction(title: "OK", style: UIAlertAction.Style.default, handler: nil))
+            self.present(alert, animated: true, completion: nil)
+        }
+    }
+
+    func validate() -> Bool {
+        if nameTextField.text! == "" || typeTextField.text! == "" || addressTextField.text! == "" || phoneTextField.text! == "" || descriptionTextView.text! == "" {
+            return false
+        }
+        return true
```
Comment (paulpop10) [inline FoodPin/FoodPin/Controller/NewRestaurantController.swift:77]: Empty space is always needed before a final return

File: FoodPin/FoodPin/Controller/NewRestaurantController.swift [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+        if let appearance = navigationController?.navigationBar.standardAppearance {
+            if let customFont = UIFont(name: "Nunito-Bold", size: 40.0) {
+                appearance.titleTextAttributes = [.foregroundColor: UIColor(named: "NavigationBarTitle")!]
+                appearance.largeTitleTextAttributes = [.foregroundColor: UIColor(named: "NavigationBarTitle")!, .font: customFont]
+            }
+
+            navigationController?.navigationBar.standardAppearance = appearance
+            navigationController?.navigationBar.compactAppearance = appearance
+            navigationController?.navigationBar.scrollEdgeAppearance = appearance
+
```
Comment (paulpop10) [inline FoodPin/FoodPin/Controller/NewRestaurantController.swift]: Unwanted empty space

File: FoodPin/FoodPin/Controller/NewRestaurantController.swift [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+        photoImageView.topAnchor.constraint(equalTo: margins.topAnchor).isActive = true
+
+        // Pin the bottom edge of the image view to the margin's bottom edge
+        photoImageView.bottomAnchor.constraint(equalTo: margins.bottomAnchor).isActive = true
+
+        // Hiding keyboard
+        let tap = UITapGestureRecognizer(target: view, action: #selector(UIView.endEditing))
+        tap.cancelsTouchesInView = false
+        view.addGestureRecognizer(tap)
+
```
Comment (paulpop10) [inline FoodPin/FoodPin/Controller/NewRestaurantController.swift]: unwanted empty space

File: FoodPin/FoodPin/Controller/NewRestaurantController.swift [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+    }
+}
+
+extension NewRestaurantController: UITextFieldDelegate {
+    func textFieldShouldReturn(_ textField: UITextField) -> Bool {
+        if let nextTextField = view.viewWithTag(textField.tag + 1) {
+            textField.resignFirstResponder()
+            nextTextField.becomeFirstResponder()
+        }
+        return true
```
Comment (paulpop10) [inline FoodPin/FoodPin/Controller/NewRestaurantController.swift:165]: Empty space needed before final return

**Review by ralumarusca — APPROVED** [review]

**Review by paulpop10 — APPROVED** [review]

**Pull request merged**